contracts/arbitrum/connectors/uniswap/v3_swap/main.sol

```diff
@@ -0,0 +1,155 @@
+//SPDX-License-Identifier: MIT
+pragma solidity ^0.7.6;
+pragma abicoder v2;
+
+/**
+ * @title Uniswap v3 swap.
+ * @dev Decentralized Exchange.
+ */
+
+import {TokenInterface} from "../../../common/interfaces.sol";
+import {Helpers} from "./helpers.sol";
+import {Events} from "./events.sol";
+import "./interface.sol";
+
+abstract contract UniswapResolver is Helpers, Events {
+	/**
+	 * @dev Buy Function
+	 * @notice Swap token(sellAddr) with token(buyAddr), buy token with minimum sell token
+	 * @param buyData Data input for the buy action
+	 */
+	function buy(
+		BuyInfo memory buyData
+	)
+		external
+		payable
+		returns (string memory _eventName, bytes memory _eventParam)
+	{
+		uint256 _buyAmt = getUint(buyData.getId, buyData.buyAmt);
+		(TokenInterface _buyAddr, TokenInterface _sellAddr) = changeEthAddress(
+			buyData.buyAddr,
+			buyData.sellAddr
+		);
+		
+		{
+			uint _slippageAmt = convert18ToDec(_sellAddr.decimals(),
+				wmul(buyData.unitAmt, convertTo18(_buyAddr.decimals(), _buyAmt))
+			);
+			require(_slippageAmt >= buyData.expectedAmt, "Too much slippage");
```

Comment:
@Richa-iitr expectedAmount should be the amount returned after the swap. It shouldn't be passed from frontend.

Reply:
Done